### PR TITLE
add template option

### PR DESCRIPTION
**Pull request by mr-tz**
`--template` option if provided creates package template files with placeholder data for non-provided values

### CODE REVIEW

File: scripts/utils/create_package_template.py [hunk 1]
```diff
@@ -419,6 +419,17 @@ def get_script_directory():
...
 
+def set_placeholder_data(args):
+    for arg in TYPES.get(args.type)["arguments"]:
+        if not arg:
+            args.__setattr__(arg, f"<{arg}>")
+        if arg == "pkg_name":
+            # default placeholder package name
+            args.__setattr__("pkg_name", "new-package")
+    sys.exit()
+    return args
```
Comment (MalwareMechanic) [inline scripts/utils/create_package_template.py:430]: Do we want to exit here?

Comment (mr-tz) [reply scripts/utils/create_package_template.py:430]: Nope, good catch. Forgot to push the last commit. Will add more documentation on this PR now.

File: scripts/utils/create_package_template.py [hunk 2]
```diff
@@ -436,30 +445,55 @@ def main(argv=None):
...
+       %(prog)s --type
+
+      Show required arguments for tool distributed in a ZIP file (ZIP_EXE template)
+       %(prog)s --type ZIP_EXE
+
+      Create package template for tool distributed in a ZIP file
+       %(prog)s --type ZIP_EXE --pkg_name new-tool --version 1.0.0 --authors 'Alice, Bob' ...
+
+      Create package template files for a METAPACKAGE using placeholder data
+       %(prog)s --type METAPACKAGE --raw
```
Comment (mr-tz) [inline scripts/utils/create_package_template.py:458]: This creates:
```
packages/pkg-name-CHANGEME.vm/tools
packages/pkg-name-CHANGEME.vm/pkg-name-CHANGEME.vm.nuspec 
packages/pkg-name-CHANGEME.vm/tools/chocolateyinstall.ps1
packages/pkg-name-CHANGEME.vm/tools/chocolateyuninstall.ps1
```

with for example `packages/pkg-name-CHANGEME.vm/pkg-name-CHANGEME.vm.nuspec`
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
  <metadata>
    <id>pkg-name-CHANGEME.vm</id>
    <version>version-CHANGEME</version>
    <authors>authors-CHANGEME</authors>
    <description>description-CHANGEME</description>
    <dependencies>
      <dependency id="common.vm" />
      <dependency id="dependency-CHANGEME" version="[version-CHANGEME]" />
    </dependencies>
  </metadata>
</package>
```

File: scripts/utils/create_package_template.py [hunk 3]
```diff
@@ -436,30 +445,55 @@ def main(argv=None):
...
+    )
+    parser.add_argument(
+        "--type",
+        required=True,
+        type=str,
+        choices=TYPES.keys(),
+        nargs="?",
+        help="Installation template type, see descriptions via %(prog)s --type",
+    )
+    parser.add_argument("--raw", action="store_true", help="Create package files like .nuspec with raw placeholder data")
```
Comment (mr-tz) [inline scripts/utils/create_package_template.py:474]: renamed this to `raw` from originally `template`

File: scripts/utils/create_package_template.py [hunk 4]
```diff
@@ -436,30 +445,55 @@ def main(argv=None):
...
+    parser.add_argument("--pkg_name", type=str, default="", help="Package name without suffix (i.e., no '.vm' needed)")
+    parser.add_argument("--version", type=str, default="", help="Tool's version number")
+    parser.add_argument("--authors", type=str, default="", help="Comma separated list of authors for tool")
+    parser.add_argument("--tool_name", type=str, default="", help="Name of tool (usually the file name with the '.exe')")
+    parser.add_argument("--category", type=str, default="", choices=CATEGORIES, help="Category for tool")
```
Comment (mr-tz) [inline scripts/utils/create_package_template.py:479]: diff due to the added `default=""`

File: scripts/utils/create_package_template.py [hunk 5]
```diff
@@ -419,6 +419,15 @@ def get_script_directory():
 }
 
 
+def set_placeholder_data(args):
+    for arg in TYPES.get(args.type)["arguments"]:
+        # set placeholder data for unset arguments
+        if not getattr(args, arg):
+            placeholder = f"{arg}-CHANGEME".replace('_', '-')
```
Comment (Ana06) [inline scripts/utils/create_package_template.py:426]: I find `CHANGEME` a bit difficult to read, I would use `CHANGE-ME` instead. I think it is also clear to apply the replace only to `arg`
```suggestion
            placeholder = f"{arg.replace('_', '-')}-CHANGE-ME"
```

File: scripts/utils/create_package_template.py [hunk 6]
```diff
@@ -436,30 +445,55 @@ def main(argv=None):
...
-    parser.add_argument("--type", type=str, choices=TYPES.keys(), nargs="?", help="Template type")
     args = parser.parse_args(args=argv)
 
     if args.type is None:
-        print(f"{'type'.ljust(15)} {'description'.ljust(60)} {'example'}")
+        print("No type provided, available options are:")
+        print(f"{'type'.ljust(15)} {'description'.ljust(62)} {'example'}")
         for k, t in TYPES.items():
-            print(f"{k.ljust(15)} {t['doc'].ljust(60)} {t['example']}")
+            print(f"{k.ljust(15)} {t['doc'].ljust(62)} {t['example']}")
```
Comment (Ana06) [inline scripts/utils/create_package_template.py:491]: why is this change needed?

**Review by Ana06 — APPROVED** [review]
This is a great idea! 👍

**Review by Ana06 — APPROVED** [review]
LGTM! Remember to squash the commits(either manually or when merging ) 😄
